packages/devextreme/js/__internal/grids/pivot_grid/remote_store/m_remote_store.ts

```diff
@@ -347,143 +363,173 @@ function getExpandedIndex(options, axis) {
-      firstCollapsedIndex = index;
-      return false;
-    }
-
-    return undefined;
-  });
-
-  return firstCollapsedIndex;
-}
-
-function getRequestsData(options) {
+function createRequestsOptions(options) {
+  const rowExpandedIndex = getExpandedIndex(options, 'rows');
   const rowExpandedLevel = getExpandedLevel(options, 'rows');
-  const columnExpandedLevel = getExpandedLevel(options, 'columns');
-  let filters = options.filters || [];
   const columnExpandedIndex = getExpandedIndex(options, 'columns');
-  const firstCollapsedColumnIndex = getFirstCollapsedIndex(options.columns);
-  const firstCollapsedRowIndex = getFirstCollapsedIndex(options.rows);
-  const rowExpandedIndex = getExpandedIndex(options, 'rows');
-  let data: any = [];
+  const columnExpandedLevel = getExpandedLevel(options, 'columns');
+
+  const requestsOptions: any = [];
 
-  filters = filters.concat(getFiltersForDimension(options.rows))
-    .concat(getFiltersForDimension(options.columns))
-    .concat(getFiltersForExpandedDimension(options));
+  const commonFilters = (options.filters || []).concat(
+    getFiltersForDimension(options.rows),
+    getFiltersForDimension(options.columns),
+    getFiltersForExpandedDimension(options),
+  );
 
-  const columnTotalsOptions = getGrandTotalRequest(options, 'columns', columnExpandedIndex, columnExpandedLevel, filters, firstCollapsedColumnIndex);
+  const columnTotalsOptions = getGrandTotalRequest(options, 'columns', commonFilters);
+  const rowTotalsOptions = getGrandTotalRequest(options, 'rows', commonFilters);
 
   if (options.rows.length && options.columns.length) {
     if (options.headerName !== 'rows') {
-      data = data.concat(columnTotalsOptions);
+      requestsOptions.push(...columnTotalsOptions);
     }
 
-    for (let i = rowExpandedIndex; i < rowExpandedLevel + 1; i += 1) {
+    for (let i = rowExpandedIndex; i <= rowExpandedLevel; i += 1) {
       const rows = options.rows.slice(rowExpandedIndex, i + 1);
-      const rowFilterByExpandedPaths = getExpandedPathSliceFilter(options, 'rows', i, firstCollapsedRowIndex);
+      const rowsFilterExpr = getExpandedPathsFilterExprByLevel(options, 'rows', i);
+
+      for (let j = columnExpandedIndex; j <= columnExpandedLevel; j += 1) {
+        const columns = options.columns.slice(columnExpandedIndex, j + 1);
+        const columnsFilterExpr = getExpandedPathsFilterExprByLevel(options, 'columns', j);
+
+        const filterExpression = mergeFilters([rowsFilterExpr, columnsFilterExpr]);
 
-      for (let j = columnExpandedIndex; j < columnExpandedLevel + 1; j += 1) {
-        const preparedOptions = extend({}, options, {
-          columns: options.columns.slice(columnExpandedIndex, j + 1),
+        const requestOptions = extend({}, options, {
+          columns,
           rows,
-          filters: filters.concat(getExpandedPathSliceFilter(options, 'columns', j, firstCollapsedColumnIndex)).concat(rowFilterByExpandedPaths),
+          filters: commonFilters,
```

Comment:
`requestOptions.filters` is an array of `fields`. `getExpandedPathSliceFilter` was extending fields with properties `field.filterValues` and 'field.filterType='include'`. `field.filterValues` is an array of values of the field that we want to request. Later this array is transformed into filters: https://github.com/DevExpress/DevExtreme/pull/28884/files#diff-b5d7b2b7d8a915d9d730cc006e8643b086ab02749e516b8cf9d46ec2dd1bd121L174

But now filters for fields are directly generated in the form of filter expression (aka `[['field', '=', 'some value']]`) and then just merged with `options.filters`: https://github.com/DevExpress/DevExtreme/pull/28884/files#diff-b5d7b2b7d8a915d9d730cc006e8643b086ab02749e516b8cf9d46ec2dd1bd121R205-R207